torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -336,6 +336,11 @@ def foreach_reduce(
                     new_sharded_grad
                 )
                 fsdp_param.sharded_param.grad = new_sharded_dtensor_grad
+            for hook in (
+                getattr(fsdp_param.sharded_param, "_post_accumulate_grad_hooks", {})
```

Comment:
Currently, `torch.Tensor`s should always have `_post_accumulate_grad_hooks` defined, regardless of if leaf or if a hook is actually registered, so the `getattr` is not necessary. This is gray area anyway since we are relying on a private attribute though.

Reply:
totally impressed by the minimal lines of change to showcase optimizer-in-the-backward